darts/metrics/metrics.py

```diff
@@ -241,35 +284,68 @@ def wrapper_multivariate_support(*args, **kwargs) -> METRIC_OUTPUT_TYPE:
             sanity_check=False,
         )
         if component_reduction is not None:
-            vals = np.expand_dims(component_reduction(vals, axis=COMP_AX), axis=COMP_AX)
+            # -> (*, n_quantile)
+            vals = component_reduction(vals, axis=COMP_AX)
+        else:
+            # -> (*, n_comp * n_quantile), with order [c0_q0, c0_q1, ... c1_q0, c1_q1, ...]
+            vals = vals.reshape(vals.shape[0], -1)
         return vals
 
     return wrapper_multivariate_support
 
 
 def _get_values(
-    vals: np.ndarray, stochastic_quantile: Optional[float] = 0.5
+    vals: np.ndarray,
+    vals_components: pd.Index,
+    actual_components: pd.Index,
+    stochastic_quantile: Optional[Tuple[List[float], Union[Optional[pd.Index]]]] = None,
 ) -> np.ndarray:
     """
     Returns a deterministic or probabilistic numpy array from the values of a time series.
     For stochastic input values, return either all sample values with (stochastic_quantile=None) or the quantile sample
-    value with (stochastic_quantile {>=0,<=1})
+    value with (quantile values {>=0,<=1})
+
+    Parameters
+    ----------
+    vals
+        A numpy array with the values of a TimeSeries (actual values or predictions).
+    vals_components
+        The components of the `vals` TimeSeries.
+    actual_components
+        The components of the actual TimeSeries.
+    stochastic_quantile
+        Optionally, a tuple with (quantile values, quantile values if `pred_series` is stochastic
+        and the quantile component names otherwise).
     """
+    if stochastic_quantile is not None:
+        q, q_names = stochastic_quantile
+    else:
+        q, q_names = None, None
+
     if vals.shape[2] == 1:  # deterministic
-        out = vals[:, :, 0]
-    else:  # stochastic
-        if stochastic_quantile is None:
-            out = vals
-        else:
-            out = np.quantile(vals, stochastic_quantile, axis=2)
-    return out
+        if isinstance(q_names, pd.Index):
+            # `q_names` are the component names of the predicted quantile parameters
+            # we extract the relevant quantile components with shape (times, components * quantiles)
+            vals = vals[:, vals_components.get_indexer(q_names)]
+            # rearrange into (times, components, quantiles)
+            vals = vals.reshape((len(vals), len(actual_components), -1))
+        return vals
+
+    # stochastic
+    if stochastic_quantile is None:
+        return vals
```

Comment:
This condition is not clear, since it's technically an `else` of the `vals.shape[2] == 1` and actually directly return the `vals` input argument, I would rather move it at the very top? Before even unpacking the `stochastic_quantile` argument.

Reply:
Good one, makes it more simple 👍